deployer/deployer-services/src/main/java/io/hyscale/deployer/services/processor/PodParentUtil.java

```diff
@@ -100,4 +107,27 @@ public PodParent getPodParentForService(String serviceName){
         }
         return Collections.emptyList();
     }
+
+    public static LoadBalancer getLoadBalancerInSpec(PodParent podParent) {
+        try {
+            V1ObjectMeta metadata = null;
+            if (ResourceKind.DEPLOYMENT.getKind().equalsIgnoreCase(podParent.getKind())) {
+                metadata = ((V1Deployment) podParent.getParent()).getMetadata();
+            }
+            if (ResourceKind.STATEFUL_SET.getKind().equalsIgnoreCase(podParent.getKind())) {
+                metadata = ((V1StatefulSet) podParent.getParent()).getMetadata();
+            }
+            if (metadata != null) {
+                JsonParser jsonParser = new JsonParser();
+                JsonObject lastAppliedConfigJson = (JsonObject) jsonParser.parse(metadata.getAnnotations().get(AnnotationKey.K8S_HYSCALE_LAST_APPLIED_CONFIGURATION.getAnnotation()));
+                JsonObject serviceSpec = (JsonObject) jsonParser.parse(lastAppliedConfigJson.getAsJsonObject("metadata").getAsJsonObject("annotations").get(AnnotationKey.HYSCALE_SERVICE_SPEC.getAnnotation()).getAsString());
+                if (serviceSpec.get("loadBalancer") != null) {
+                    return GsonProviderUtil.getPrettyGsonBuilder().fromJson(serviceSpec.get("loadBalancer"), LoadBalancer.class);
+                }
```

Comment:
Do we have a way to fetch load balancer from the cluster itself? So that we don't have to depend on pod parent?

Reply:
In the cluster, we may not have the details of the input hspec. Here we need hspec to identify whether the hspec has load balancer snippet or not.